src/Language/Haskell/Ghcid.hs

```diff
@@ -94,4 +105,9 @@ stopGhci ghci = handle (\UnexpectedExit{} -> return ()) $ void $ exec ghci ":qui
 
 -- | Send a command, get lines of result
 exec :: Ghci -> String -> IO [String]
-exec (Ghci x) = x
+exec (Ghci (_,f)) = f
+
+-- | Interrupt the test command.
+interrupt :: Ghci -> Maybe String -> Bool -> IO ()
+interrupt (Ghci (ph,_)) test spawn =
+    when (isJust test && spawn) $ interruptProcessGroupOf ph
```

Comment:
The API should be `interrupt :: Ghci -> IO ()` and the checks should be in the non-API half.